controllers/cloud.redhat.com/providers/deployment/utils.go

```diff
@@ -0,0 +1,274 @@
+package deployment
+
+import (
+	"fmt"
+
+	crd "cloud.redhat.com/clowder/v2/apis/cloud.redhat.com/v1alpha1"
+	"cloud.redhat.com/clowder/v2/controllers/cloud.redhat.com/utils"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func GetDeploymentName(app *crd.ClowdApp, deployment *crd.Deployment) string {
+	return fmt.Sprintf("%s-%s", app.Name, deployment.Name)
+}
+
+func (dp *deploymentProvider) makeDeployment(deployment crd.Deployment, app *crd.ClowdApp) error {
+
+	d := &apps.Deployment{}
+	nn := types.NamespacedName{
+		Name:      GetDeploymentName(app, &deployment),
+		Namespace: app.Namespace,
+	}
+
+	if err := dp.Cache.Create(CoreDeployment, nn, d); err != nil {
+		return err
+	}
+
+	initDeployment(app, dp.Env, d, nn, deployment)
+
+	if err := dp.Cache.Update(CoreDeployment, d); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func initDeployment(app *crd.ClowdApp, env *crd.ClowdEnvironment, d *apps.Deployment, nn types.NamespacedName, deployment crd.Deployment) {
+	labels := app.GetLabels()
+	labels["pod"] = nn.Name
+	app.SetObjectMeta(d, crd.Name(nn.Name), crd.Labels(labels))
+
+	pod := deployment.PodSpec
+
+	d.Spec.Replicas = deployment.MinReplicas
+	d.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+	d.Spec.Template.ObjectMeta.Labels = labels
+	d.Spec.Strategy = apps.DeploymentStrategy{
+		Type: apps.RollingUpdateDeploymentStrategyType,
+		RollingUpdate: &apps.RollingUpdateDeployment{
+			MaxSurge:       &intstr.IntOrString{Type: intstr.Int, IntVal: int32(25)},
+			MaxUnavailable: &intstr.IntOrString{Type: intstr.Int, IntVal: int32(25)},
+		},
+	}
+	d.Spec.ProgressDeadlineSeconds = utils.Int32Ptr(600)
+
+	d.Spec.Template.Spec.ImagePullSecrets = []core.LocalObjectReference{
+		{Name: "quay-cloudservices-pull"},
+	}
+
+	d.Spec.Template.Spec.ServiceAccountName = app.GetClowdSAName()
```

Comment:
should this be set in the service account provider?

Reply:
Yes - this is happening in other places too - we should certainly deal with this......probably in this PR, but we need to sort out how to pull these deployments out